### DISCUSSION TITLE
Closed kinematic chain simulation

### DISCUSSION THREAD

**Discussion by finnfi** [google/brax#416] [General]
Hi, 
We are currently trying to setup a simulation of a quadruped where each leg is designed like a five-bar-linkage system like this: 
![image](https://github.com/google/brax/assets/32849595/1240d317-65ca-41bb-a6f2-27e4525afc1a)

Is this closed chain kinematics supported by brax?

**Comment by btaba** [comment]
Hi @finnfi , we don't model closed kinematic chains in brax, but the way to do this in MuJoCo (or for the generalized backend) is to add a [weld constraint](https://mujoco.readthedocs.io/en/stable/XMLreference.html#equality-weld) at the paw. We don't have weld implemented in generalized right now
For spring/pbd, I would (maybe) expect such a system to sort of work (with a lot of tuning). But since we use the MJCF loader, you won't be able to create a system with a kinematic loop in Brax.
To sum it up, we'd need weld constraints implemented in Brax
